### PR TITLE
flagmodify: Remove the use of flags for all logging options

### PR DESCRIPTION
**Pull request by llukalapu-uptycs**
https://uptycsjira.atlassian.net/browse/CS-1045

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,22 +10,23 @@
...
   resource_prefix = "cloudquery"
 
   # These two values are provided by Uptycs
   aws_account_id = "1234567890"
   external_id    = "f09cd4ae-76f1-4373-88da-de721312803d"
 
-  # Pass bucket names if cloud_logs_enabled = true
-  cloud_logs_enabled        = false
-  vpc_flowlogs_bucket_name  = ""
+  # Choose one option between cloudtrail bucket and kinesis stream for cloud logs
```
Comment (apatil-uptycs) [inline README.md:20]: cloud trail instead of cloudlog

File: README.md [hunk 2]
```diff
@@ -10,22 +10,23 @@
...
+
+  # Pass bucket name if customer wants to attach bucket for cloudlogs
```
Comment (apatil-uptycs) [inline README.md:22]: cloudtrail

File: README.md [hunk 3]
```diff
@@ -10,22 +10,23 @@
...
   cloudtrail_s3_bucket_name = ""
 
-  # Pass kinesis stream name if kinesis_stream_enabled = true
-  kinesis_stream_enabled = false
-  kinesis_stream_name    = ""
+  # Pass kinesis stream name if customer wants to attach kinesis stream for cloudlogs
```
Comment (apatil-uptycs) [inline README.md:25]: cloudtrail

File: README.md [hunk 4]
```diff
@@ -45,11 +46,9 @@ output "aws-iam-role-arn" {
...
-| vpc_flowlogs_bucket_name  | S3 bucket where VPC flow logs are saved. Required if cloud_logs_enabled is set to 'true'               | `string` | `""`         |
-| cloudtrail_s3_bucket_name | S3 bucket where CloudTrail is saved. Requried if cloud_logs_enabled is set to 'true'                   | `string` | `""`         |
-| kinesis_stream_enabled    | This is set true or false i.e. whether you want to use kinesis stream or not                           | `bool`   | `false`      |
-| kinesis_stream_name       | Kinesis stream where CloudTrail logs are streamed. Required if kinesis_stream_enabled is set to 'true' | `string` | `""`         |
+| vpc_flowlogs_bucket_name  | S3 bucket where VPC flow logs are saved. Required, if you want to use S3 bucket for vpc flow logs               | `string` | `""`         |
+| cloudtrail_s3_bucket_name | S3 bucket where CloudTrail is saved. Requried, if you want to use S3 bucket for cloud logs                  | `string` | `""`         |
+| kinesis_stream_name       | Kinesis stream where CloudTrail logs are streamed. Required, if you want to use kinesis stream for cloud logs | `string` | `""`         |
```
Comment (apatil-uptycs) [inline README.md:51]: for cloudtrail

File: README.md [hunk 5]
```diff
@@ -77,5 +76,6 @@ $ terraform apply
 - The user should have `Administrators` role permission to create resources.
 - If the user has multiple aws account profiles then set profile before execute terraform.
   ```sh
-     export AWS_PROFILE="< profile name >"
+    export AWS_PROFILE="< profile name >"
   ```
+ - In file.tf file, choose one option between cloudtrail bucket and kinesis stream for cloudlog monitoring . Give priority to kinesis stream
```
Comment (apatil-uptycs) [inline README.md:81]: cloudtrail monitoring

**Comment by apatil-uptycs** [comment]
@chandrakanta-uptycs Can you please review this ?

File: README.md (hunk 1 above)
Comment (spasam) [inline README.md:20]:
```suggestion
  # Choose S3 bucket or Kinesis stream option to ingest CloudTrail 
```

File: README.md (hunk 2 above)
Comment (spasam) [inline README.md:22]:
```suggestion
  # Provide the S3 bucket name which contains the CloudTrail data
```

File: README.md (hunk 3 above)
Comment (spasam) [inline README.md:25]:
```suggestion
  # Name of the Kinesis stream configured to stream CloudTrail data
```

File: README.md [hunk 6]
```diff
@@ -10,22 +10,23 @@
...
+  # Pass bucket name if customer wants to attach bucket for cloudlogs
   cloudtrail_s3_bucket_name = ""
 
-  # Pass kinesis stream name if kinesis_stream_enabled = true
-  kinesis_stream_enabled = false
-  kinesis_stream_name    = ""
+  # Pass kinesis stream name if customer wants to attach kinesis stream for cloudlogs
+  kinesis_stream_name = ""
+
+  # Pass bucket name if customer wants to attach bucket for vpc flow logs
```
Comment (spasam) [inline README.md:28]:
```suggestion
  # Name of the S3 bucket that contains the VPC flow logs
```

File: README.md [hunk 7]
```diff
@@ -45,11 +46,9 @@ output "aws-iam-role-arn" {
...
-| cloud_logs_enabled        | This is set true or false i.e. whether you want to use log buckets or not                              | `bool`   | `false`      |
-| vpc_flowlogs_bucket_name  | S3 bucket where VPC flow logs are saved. Required if cloud_logs_enabled is set to 'true'               | `string` | `""`         |
-| cloudtrail_s3_bucket_name | S3 bucket where CloudTrail is saved. Requried if cloud_logs_enabled is set to 'true'                   | `string` | `""`         |
-| kinesis_stream_enabled    | This is set true or false i.e. whether you want to use kinesis stream or not                           | `bool`   | `false`      |
-| kinesis_stream_name       | Kinesis stream where CloudTrail logs are streamed. Required if kinesis_stream_enabled is set to 'true' | `string` | `""`         |
+| vpc_flowlogs_bucket_name  | S3 bucket where VPC flow logs are saved. Required, if you want to use S3 bucket for vpc flow logs               | `string` | `""`         |
+| cloudtrail_s3_bucket_name | S3 bucket where CloudTrail is saved. Requried, if you want to use S3 bucket for cloud logs                  | `string` | `""`         |
```
Comment (spasam) [inline README.md:50]: Same as above

File: README.md (hunk 4 above)
Comment (spasam) [inline README.md:51]: Replace description with what I suggested above

File: README.md (hunk 5 above)
Comment (spasam) [inline README.md:81]:
```suggestion
 - In file.tf file, specify CloudTrail S3 bucket name or Kinesis stream name. Kinesis stream based approach provides faster CloudTrail data ingestion
```

File: modules/iam-profile/main.tf [hunk 8]
```diff
@@ -72,7 +72,20 @@ resource "aws_iam_policy" "ReadOnlyPolicy" {
...
+              "eks:ListAddons",
+              "eks:DescribeAddon",
+              "eks:ListFargateProfiles",
+              "eks:DescribeNodegroup",
+              "eks:DescribeIdentityProviderConfig",
+              "eks:ListUpdates",
+              "eks:DescribeUpdate",
+              "eks:ListIdentityProviderConfigs",
+              "servicecatalog:DescribeServiceAction",
+              "servicecatalog:DescribeProvisioningArtifact"
```
Comment (spasam) [inline modules/iam-profile/main.tf:88]: Where are these coming from? And we should keep these alphabetically ordered

Comment (llukalapu-uptycs) [reply modules/iam-profile/main.tf:88]: @spasam 
based on this https://github.com/Uptycs/uptycs/commit/3dee14328f6df1145960de1d26099763ac5f13f2 i added these permissions

File: main.tf [hunk 9]
```diff
@@ -1,27 +1,25 @@
 data "aws_s3_bucket" "vpc_log_bucket_arn" {
```
Comment (chandrakanta-uptycs) [inline main.tf:1]: If cloud_logs_enabled flag is there then i think we don't need data section .

Comment (chandrakanta-uptycs) [reply main.tf:1]: ohh ..got it ...

File: modules/iam-profile/main.tf [hunk 10]
```diff
@@ -85,16 +114,16 @@ EOF
...
   role       = aws_iam_role.role.name
   policy_arn = aws_iam_policy.cloud_trail_bucketPolicy[0].arn
 
 }
 
 resource "aws_iam_policy" "cloud_trail_bucketPolicy" {
-  # Only required when cloud logs are enabled
-  count       = var.cloud_logs_enabled ? 1 : 0
+  #  Only required when customer wants to  attach the bucket for cloudtrail logs
+  count       = var.cloudtrail_log_bucket_arn != null ? 1 : 0
```
Comment (chandrakanta-uptycs) [inline modules/iam-profile/main.tf:126]: count       =    var.cloudtrail_log_bucket_arn ? 1 : 0

File: modules/iam-profile/main.tf [hunk 11]
```diff
@@ -114,16 +143,16 @@ EOF
...
   role       = aws_iam_role.role.name
   policy_arn = aws_iam_policy.VpcFlowLogBucketPolicy[0].arn
 
 }
 
 resource "aws_iam_policy" "VpcFlowLogBucketPolicy" {
-  # Only required when cloud logs are enabled
-  count       = var.cloud_logs_enabled ? 1 : 0
+  # Only required when customer wants to  attach the bucket for vpc flow logs
+  count       = var.vpc_log_bucket_arn != null ? 1 : 0
```
Comment (chandrakanta-uptycs) [inline modules/iam-profile/main.tf:155]: same as above

File: modules/iam-profile/main.tf [hunk 12]
```diff
@@ -143,16 +172,16 @@ EOF
 }
 
 resource "aws_iam_role_policy_attachment" "kinesis_stream_policy_attach" {
-  # Only required when kinesis stream for cloudtrail logs is enabled
-  count      = var.kinesis_stream_enabled ? 1 : 0
+  # Only required when customer wants to  attach the kinesis stream for cloudtrail logs
+  count      = var.kinesis_stream_arn != null ? 1 : 0
```
Comment (chandrakanta-uptycs) [inline modules/iam-profile/main.tf:176]: same her as well

**Review by chandrakanta-uptycs — APPROVED** [review]

**Pull request merged**